src/model/dataclass_with_properties.py

```diff
@@ -1,24 +1,28 @@
+from dataclasses import dataclass
 from typing import Type, Any
+from typeguard import typechecked
 
 
 def dataclass_with_properties(cls):
-    """decorator to generate properties out of the dataclass value:type list.
-    Their getters and setters will be subjected to the @typechecked decorator"""
-    for field_name, field_type in cls.__annotations__.items():
+    """Decorator to generate a dataclass with properties out of the class' value:type list.
+    Their getters and setters will be subjected to the @typechecked decorator to ensure type conformity."""
+    data_cls = dataclass(cls)
+    for field_name, field_type in data_cls.__annotations__.items():
 
         set_field = make_setter(field_name, field_type)
         get_field = make_getter(field_name, field_type)
 
-        setattr(cls, f"set_{field_name}", set_field)
-        setattr(cls, f"get_{field_name}", get_field)
+        setattr(data_cls, f"set_{field_name}", set_field)
+        setattr(data_cls, f"get_{field_name}", get_field)
```

Comment:
nit: I'm pretty sure I verified that you don't actually need to set the `set_x` and `get_x` methods on the class. You can simply use the getter and setter that were defined above directly. Doesn't really matter either way, though.